ocular/src/chain/client/query.rs

```diff
@@ -20,31 +20,43 @@ pub type SlashingQueryClient = slashing::query_client::QueryClient<Channel>;
 pub type StakingQueryClient = staking::query_client::QueryClient<Channel>;
 
 impl ChainClient {
-    pub fn check_for_grpc_address(&self) -> Result<(), GrpcError> {
-        if self.config.grpc_address.is_empty() {
-            return Err(GrpcError::MissingEndpoint(format!(
-                "no grpc address available for chain {}",
-                self.config.chain_id
-            )));
+    // Auth queries
+    pub async fn get_auth_query_client(&mut self) -> Result<AuthQueryClient, ChainClientError> {
+        let mut result: Result<AuthQueryClient, ChainClientError> =
```

Comment:
because of the way the loop is written (will always attempt at least once) i don't think this can happen?

Reply:
Ah yeah this can never get returned, however I just needed to assign something to result initially; I don't think rust allows var creations without assignments, right?

Reply:
It does; you can do 

```rust
let mut result: Result<AuthQueryClient, ChainClientError>;
```

Reply:
Tried this out, build failed due to it not being initialized.

```
error[E0381]: use of possibly-uninitialized variable: `result`
```